### DISCUSSION TITLE
Typescript-json > zod ?

### DISCUSSION THREAD

**Discussion by thelinuxlich** [ecyrbe/zodios#174] [General]
There is a new player in the field of TS runtime validation libraries. [typescript-json](https://github.com/samchon/typescript-json) adds a plugin transform to tsconfig.json and can work with existing TS types.

Pros
- No need to add a extra DSL for runtime, just use the TS interfaces/types
- Supports anything fancy in TS level
- Much faster than zod
- Has a "comment tag" API for richer validation
- Might simplify zodios internals

Cons
- Doesn't have transforms
- Zod has better error output

I understand the library is deeply committed to zod but anyway I thought it would be interesting to share this, might be gamechanging.

**Comment by ecyrbe** [comment]
Hello, 

please do not create issues for questions. Go on discussions tab, it's made for this use case. I converted this to one.

For the question itself, Typescript json is based on a superset of typescript:  `ttypescript` that allows typescript to emit code.
not everybody want to have to deal with a superset of typescript and the issues it will bring.

Zodios is completely based on zod, hence the name. It's indeed a deliberate choice. 
i don't plan to migrate or create a lib based on typescript-json anytimesoon.

Indeed, Zodios is now javascript compliant library (meaning it works and autocompletes your code even though you are not using typescript). 
It will stay aligned with this promise and guide every future design of the library.

Thank you for asking.

**Comment by ecyrbe** [comment]
@thelinuxlich With the new type provider system, i guess you'll now be able to create a provider for typescript-json
